libvast/vast/concept/parseable/vast/type.hpp

```diff
@@ -62,141 +60,30 @@ class type_table : public parser<type_table> {
   symbol_table<type> symbols_;
 };
 
-/// Parses a type with the help of a symbol table.
+/// Parses a type into an intermediate representation.
+/// References to user defined types are mapped to `none_type` and
+/// need to be resolved later.
 struct type_parser : parser<type_parser> {
   using attribute = type;
 
   // Comments until the end of line.
   // clang-format off
   static constexpr auto comment
     = ignore(parsers::lit{"//"} >> *(parsers::any - '\n'));
+
+  /// The parser for an identifier.
+  static constexpr auto id
+    = +( parsers::alnum
+       | parsers::ch<'_'>
+       | parsers::ch<'.'>
+       );
```

Comment:
IIRC we have another parser for this? If not, it might be worthwhile factoring.

(Maybe not in this PR.)